packages/firebase_messaging/README.md

```diff
@@ -98,19 +99,48 @@ By default background messaging is not enabled. To handle messages in the backgr
    }
    ```
 
-1. In `Application.java`, make sure to change `package io.flutter.plugins.firebasemessagingexample;` to your package's identifier. Your package's identifier should be something like `com.domain.myapplication`.
+3. In `Application.java`, make sure to change `package io.flutter.plugins.firebasemessagingexample;` to your package's identifier. Your package's identifier should be something like `com.domain.myapplication`.
 
    ```java
    package com.domain.myapplication;
    ```
+#### Kotlin's support:
+2. Add an `Application.kt` class to your app in the same directory as your `MainActivity.kt`. This is typically found in `<app-name>/android/app/src/main/kotlin/<app-organization-path>/`.
+```kotlin
+package io.flutter.plugins.firebasemessagingexample
+
+import io.flutter.app.FlutterApplication
+import io.flutter.plugin.common.PluginRegistry
+import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
+import io.flutter.plugins.GeneratedPluginRegistrant
+import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
+
+class Application : FlutterApplication(), PluginRegistrantCallback {
+
+    override fun onCreate() {
+        super.onCreate()
+        FlutterFirebaseMessagingService.setPluginRegistrant(this);
+    }
+
+    override fun registerWith(registry: PluginRegistry?) {
+        io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin.registerWith(registry?.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"));
+    }
+}
+```
+3. In `Application.kt`, make sure to change `package io.flutter.plugins.firebasemessagingexample` to your package's identifier. Your package's identifier should be something like `com.domain.myapplication`.
+
+   ```kotlin
+   package com.domain.myapplication;
+   ```
+---
 
-1. Set name property of application in `AndroidManifest.xml`. This is typically found in `<app-name>/android/app/src/main/`.
+4. Set name property of application in `AndroidManifest.xml`. This is typically found in `<app-name>/android/app/src/main/`.
 
    ```xml
    <application android:name=".Application" ...>
```

Comment:
It would be better to show from and to for people not familiar with editing AndroidManifest.xml files.

```suggestion
   <!-- FROM -->
	<application android:name="io.flutter.app.FlutterApplication" ... />
	
	<!-- TO -->
	<application android:name=".Application" ... />
```